ONSdigital/dp-frontend-search-controller
# Fix submit not working as expected

### What

Link to Jira ticket: https://jira.ons.gov.uk/browse/DIS-1568

To see the current problems first go to the 'Published data' list here: https://www.ons.gov.uk/datalist

Problem 1:
Entering a word (e.g. "cheese") into the 'Search keywords' text box, on the 'Published data' list, currently gives a URL containing many parameters and two of them are named "q":

https://www.ons.gov.uk/datalist?q=cheese&after-day=&after-month=&after-year=&before-day=&before-month=&before-year=&q=&sort=release_date

This change removes the "q" parameter from the right hand side, as it is not being used for anything, and renames the "q" on the left to be called "kw-query". This is because there are already several other "q" parameters in use on the same page.

Problem 2: 
Ticking 2 of the filter check boxes on the left (e.g. 'Time series' and 'Datasets') then clicking the green button gives you an initial URL like this, which is correct so far:

https://www.ons.gov.uk/datalist?q=&filter=time_series&filter=datasets&after-day=&after-month=&after-year=&before-day=&before-month=&before-year=&q=&sort=release_date

If you then untick the 'Datasets' filter (for example) and click the green button again then its check box is automatically re-ticked and both filters are incorrectly re-added to the right hand side of the URL, like this:

ttps://www.ons.gov.uk/datalist?q=&filter=time_series&after-day=&after-month=&after-year=&before-day=&before-month=&before-year=&q=&filter=datasets&filter=time_series&sort=release_date

### How to review

In your local environment do as follows:

- Run the dp-frontend-search-controller:
`make debug API_ROUTER_URL=https://api.dp.aws.onsdigital.uk/v1 ENABLE_AGGREGATION_PAGES=true`
- Run the dp-frontend-router:
`make debug API_ROUTER_URL=https://api.dp.aws.onsdigital.uk/v1 SEARCH_ROUTES_ENABLED=true DATA_AGGREGATION_PAGES_ENABLED=true`
- Run the dp-design-system:
`make debug`
- You should then see the sandbox list of published data here:
http://localhost:20000/datalist
- Enter a word to search for in the 'Search keywords' textbox e.g. Cheese
- Click the green button next to the textbox
- Observe that the URL should now look something like this (problem 1 is fixed):

http://localhost:20000/datalist?kw-query=cheese&after-day=&after-month=&after-year=&before-day=&before-month=&before-year=&sort=release_date

- Tick the 'Time series' and 'Datasets' check boxes, click the green button again, to change the URL to this:

http://localhost:20000/datalist?kw-query=cheese&filter=time_series&filter=datasets&after-day=&after-month=&after-year=&before-day=&before-month=&before-year=&sort=release_date

- Untick 'Datasets' and click the green button again. Observe that 'Datasets' remains unticked and is removed from the URL (problem 2 is fixed):

http://localhost:20000/datalist?kw-query=cheese&filter=time_series&after-day=&after-month=&after-year=&before-day=&before-month=&before-year=&sort=release_date

### Who can review

Anyone.
